lib/Map/RegionProvider.js

```diff
@@ -200,39 +240,32 @@ RegionProvider.prototype.loadRegionIDs = function() {
         throw (new DeveloperError('No server for region mapping defined: ' + this.regionType));
     }
     var that = this;
-
-    var baseuri = URI(this.server).addQuery({
-        service: 'wfs',
-        version: '2.0',
-        request: 'getPropertyValue',
-        typenames: this.layerName});
-
-    // get the list of IDs that we will attempt to match against for this column
-    var promises = [];
-    var url = baseuri.setQuery('valueReference', this.regionProp).toString();
-
-    if (corsProxy.shouldUseProxy(url)) {
-        url = corsProxy.getURL(url);
+    var p;
+    if (defined(this.regionIdList)) {
+        p = loadJson(this.regionIdList);
+    } else {
+        p = this.loadRegionsFromWfs(this.regionProp);
     }
+    p.then(function(json) {
+        return that.processRegionIds(json.values, undefined, 'serverReplacements');
+    }).otherwise(function(err) {
+        console.log(err);
+        throw(err);
+    });
+    var promises = [p];
 
-    promises.push(loadText(url)
-        .then(function(xml) {
-            that.loadRegionsFromXML(xml, undefined, "serverReplacements");
-        }).otherwise(function(err) {
-            console.log(err);
-            throw(err);
-        }));
     // if this column might be ambiguous then fetch the disambiguating values for each column as well (usually State)
     if (this.disambigProp) {
-        url = baseuri.setQuery('valueReference', this.disambigProp).toString();
-        promises.push(loadText(url).then(function(xml) {
-            that.loadRegionsFromXML(xml, that.disambigProp, "disambigServerReplacements");
-        }));
+        var dp = this.loadRegionsFromWfs(this.disambigProp);
```

Comment:
Oh yeah, now I remember - as you pointed out, "RegionProvider would need to know things about other regions". In a previous iteration of this RegionProvider thing, each RP actually did contain a reference to a whole separate RP, but that turned out to be a pretty flakey concept. A disambiguation field is just an additional field of the same boundary layer, many of the other attributes don't apply.

Fixed by adding regionDisambigIdList field, and cleaning up the code some more.